### PR TITLE
Add setuptools_scm and pip check

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,15 +22,18 @@ requirements:
   host:
     - python >=3.7
     - pip
+    - setuptools_scm
```
Comment (xylar) [inline recipe/meta.yaml:25]: This isn't listed as a build dependency but this line suggests that it's probably needed:
https://github.com/astanin/python-tabulate/blob/v0.9.0/pyproject.toml#L37

**Comment by xylar** [comment]
@synapticarbors and @jakirkham, please have a look when you can.

File: recipe/meta.yaml (hunk 1 above)
Comment (synapticarbors) [reply recipe/meta.yaml:25]: Yeah, I overlooked it in the build system section:
https://github.com/astanin/python-tabulate/blob/master/pyproject.toml#L2

**Comment by jakirkham** [comment]
Thanks Xylar! 🙏

Do the previous packages need to be [marked broken]( https://github.com/conda-forge/admin-requests#mark-packages-as-not-broken-on-conda-forge )?

File: recipe/meta.yaml (hunk 1 above)
Comment (xylar) [reply recipe/meta.yaml:25]: grayskull seems to have missed it, too.  That's disappointing.

Comment (xylar) [reply recipe/meta.yaml:25]: https://github.com/conda-incubator/grayskull/issues/340

Comment (xylar) [reply recipe/meta.yaml:25]: Looks like `pyproject.toml` just isn't fully supported by grayskull yet.

**Comment by xylar** [comment]
@jakirkham, I wouldn't bother unless we run into trouble.  I got the new build just fine just now when I tested.

**Comment by xylar** [comment]
And thank you both, @jakirkham and @synapticarbors!

**Comment by synapticarbors** [comment]
@xylar -- thanks for catching it. This is not the first time I've seen setuptools_scm get missed where it builds seemingly fine, but then the version is broken. I've also seen strange stuff where the build system requires something like flit, but then it builds fine without it.

**Comment by xylar** [comment]
I've seen this once before, though I couldn't begin to guess where.  It happens when the only thing `setuptools_scm` is used for is reading the version number from some file.

**Pull request merged**